zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -462,12 +461,14 @@ object JsonCodec {
     private def enumDecoder[Z](parentSchema: Schema.Enum[Z], cases: Schema.Case[Z, _]*): ZJsonDecoder[Z] = {
       (trace: List[JsonError], in: RetractReader) =>
         {
-          val caseNameAliases = cases.map {
+          val caseNameAliases = cases.flatMap {
             case Schema.Case(name, _, _, _, _, annotations) =>
-              (name, annotations.collectFirst {
-                case a: caseNameAliases => a.aliases.toList
-                case cn: caseName       => List(cn.name)
-              }.getOrElse(List.empty))
+              annotations.flatMap {
+                case a: caseNameAliases => a.aliases.toList.map(_ -> name)
+                case cn: caseName       => List(cn.name -> name)
+                case _                  => Nil
+              }
+            case _ => Nil
           }.toMap
```

Comment:
Inverting the `Map` to be `caseName -> name`, results in a possibly larger map but the `deAliasCaseName` will now be `O(1)`.